snap/squashfs/squashfs.go

```diff
@@ -57,12 +58,14 @@ func New(snapPath string) *Snap {
 var osLink = os.Link
 var cmdutilCommandFromSystemSnap = cmdutil.CommandFromSystemSnap
 
-func (s *Snap) Install(targetPath, mountDir string) error {
+func (s *Snap) Install(targetPath, mountDir string) (*undo_context.InstallUndoContext, error) {
```

Comment:
I think this should just return a flag whether it was a noop or not, undo etc should be modeled in snapstate/backend and not here

Reply:
Ok, I think it makes sense in general; I did that, but I'm not happy about the details yet, the "jump" from "nothing to do" to the undo context -> keep target flag snap looks a bit wrong, I need to think about it (some renaming may be needed, at the very least).